Description: Adjust Lua plugin buildscript.
 In Debian CFLAGS and LIBS for linking with Lua implementations could be got
 through pkg-config.
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Forwarded: not-needed
Last-Update: 2011-05-17

Index: uwsgi/plugins/lua/uwsgiplugin.py
===================================================================
--- uwsgi.orig/plugins/lua/uwsgiplugin.py	2011-05-15 22:40:27.000000000 +0300
+++ uwsgi/plugins/lua/uwsgiplugin.py	2011-05-15 22:41:32.000000000 +0300
@@ -1,5 +1,12 @@
 import os,sys
 
+LUAIMPL = os.environ['UWSGICONFIG_LUA_IMPLEMENTATION']
+CFLAGS = os.popen('pkg-config --cflags %s' % LUAIMPL).read().rstrip().split()
+LDFLAGS = []
+LIBS = os.popen('pkg-config --libs %s' % LUAIMPL).read().rstrip().split()
+
+"""Upstream way of getting CFLAGS and LIBS was:
+
 try:
         LUALIB = os.environ['UWSGICONFIG_LUALIB']
 except:
@@ -15,8 +22,10 @@
 except:
         LUALIBPATH = '/usr/lib/lua5.1'
 
-NAME='lua'
 CFLAGS = ['-I%s' % LUAINC]
 LDFLAGS = ['-L%s' % LUALIBPATH]
-GCC_LIST = ['lua_plugin']
 LIBS = ['-l%s' % LUALIB]
+"""
+
+NAME='lua'
+GCC_LIST = ['lua_plugin']
